fdbclient/admin_client.go

```diff
@@ -318,7 +309,30 @@ func (client *cliAdminClient) getStatus(useClientLibrary bool) (*fdbv1beta2.Foun
 	if err != nil {
 		return nil, err
 	}
-	client.log.V(1).Info("Fetched status JSON", "status", status)
+
+	client.log.V(1).Info("Fetched status JSON with fdbcli", "version", client.Cluster.GetRunningVersion(), "status", status)
+
+	return status, nil
+}
+
+// getStatus uses fdbcli to connect to the FDB cluster, if the cluster is upgraded and the initial version returns no processes
+// the new version for fdbcli will be tried.
+func (client *cliAdminClient) getStatus() (*fdbv1beta2.FoundationDBStatus, error) {
+	status, err := client.getStatusFromCli()
+	if err != nil {
+		return nil, err
+	}
+
+	// If the status doesn't contain any processes and we are doing an upgrade, we probably use the wrong fdbcli version
+	// and we have to fallback to the on specified in out spec.version.
+	if len(status.Cluster.Processes) == 0 && client.Cluster.IsBeingUpgraded() {
```

Comment:
It looks like this has less logging in this scenario than the existing code, which worries me because of how that logging has been helpful in recent debugging. Can we add that logging back? We might even want to expand it to include the invalid status we got from the client library to give more context.

Reply:
I don't think that this scenario has less logging than before. We still report the `"Fetched status JSON with fdbcli"` log line including the used version. In addition we still log the command that is executed including the full binary path and the timeout parameter and I haven't removed any log statements. What logging statement exactly are you missing/looking for?

Reply:
I added some additional logging.